src/models/itemFilterData.ts

```diff
@@ -0,0 +1,72 @@
+import type {
+  ItemAggregation,
+  ItemAggregationOption,
+  Option,
+} from "../types/filterTypes"
+
+import { isRecapLocation } from "../utils/itemFilterUtils"
+
+export class ItemFilterData {
+  options: ItemAggregationOption[]
+  agg: ItemAggregation
+  field: string
+  constructor(agg: ItemAggregation) {
+    this.agg = agg
+    this.options = agg.values
+    this.field = agg.field
+  }
+
+  displayOptions(): Option[] {
+    return this.options
+  }
+
+  labelForValue(value: string) {
+    return this.displayOptions().find((opt: Option) => {
+      return opt.value === value
+    })?.label
+  }
+
+  labelsForConcatenatedValues(values: string) {
+    return Array.from(
+      new Set(
+        values.split(",").map((val: string) => `'${this.labelForValue(val)}'`)
+      )
+    ).join(", ")
+  }
+}
+
+export class LocationFilterData extends ItemFilterData {
```

Comment:
I wonder if you need tests for these classes when it's pretty much covered in the other tests. What do you think? Or perhaps a later PR?

Reply:
I'm on the fence, too, which is why I asked. I think integration is fine since it covers a lot but now that we're including tests and I see a file without one, it makes me _feel_ like that's an issue. But it's really not.